proxy_agent/src/key_keeper/key.rs

```diff
@@ -629,112 +627,88 @@ impl Clone for Key {
     }
 }
 
-const STATUS_URL: &str = "/secure-channel/status";
-const KEY_URL: &str = "/secure-channel/key";
+const STATUS_URL: &str = "secure-channel/status";
+const KEY_URL: &str = "secure-channel/key";
 
 // base_url must end with '/'
-pub fn get_status(base_url: Url) -> std::io::Result<KeyStatus> {
-    let url = base_url.join(STATUS_URL).unwrap();
-    let mut req = Request::new(STATUS_URL.to_string(), "GET".to_string());
-    req.headers
-        .add_header(constants::METADATA_HEADER.to_string(), "True ".to_string());
-    req.headers.add_header(
+pub async fn get_status(base_url: Url) -> std::io::Result<KeyStatus> {
+    let url = format!("{}{}", base_url, STATUS_URL);
+    let mut headers = HashMap::new();
+    headers.insert(constants::METADATA_HEADER.to_string(), "True ".to_string());
+    headers.insert(
         constants::DATE_HEADER.to_string(),
         misc_helpers::get_date_time_rfc1123_string(),
     );
-    let mut http_request = HttpRequest::new(url, req);
-    http_request
-        .request
-        .headers
-        .add_header("Host".to_string(), http_request.get_host());
-
-    let response = http::get_response_in_string(&mut http_request)?;
-    if response.status != Response::OK {
-        return Err(Error::new(
-            ErrorKind::Other,
-            format!("Host response {}", response.status),
-        ));
-    }
-
-    let status: KeyStatus = serde_json::from_str(&response.get_body_as_string()?)?;
+    let status: KeyStatus = http::get(&url, &headers, None, None, logger::write_warning).await?;
     status.validate()?;
 
     Ok(status)
 }
 
 // base_url must end with '/'
```

Comment:
Given that these functions expect `base_url` to end with '/', can we just add an assertion that checks if it does in the function call?

Reply:
make sense, I use the url.join to generate the url string and caught the parse-errors.